### PR TITLE
Add tests for derived data pr

### PR DESCRIPTION
**Pull request by i2amsam**
Adds a quick test for https://github.com/fastlane/fastlane/pull/3633

### CODE REVIEW

File: snapshot/spec/test_command_generator_spec.rb [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
+
+      context 'fixed derivedDataPath' do
+        before do
+          configure options.merge(derived_data_path: 'fake/derived/path')
+        end
+
+        it 'uses the fixed derivedDataPath if given' do
+          expect(Dir).not_to receive(:mktmpdir)
+          command = Snapshot::TestCommandGenerator.generate(device_type: "Something")
+          expect(command.join('')).to include("-derivedDataPath 'fake/derived/path'")
```
Comment (mfurtak) [inline snapshot/spec/test_command_generator_spec.rb]: Seems inconsequential, but does `join(' ')` make a bit more sense?

File: snapshot/spec/test_command_generator_spec.rb [hunk 2]
```diff
@@ -1,26 +1,50 @@
+require 'benchmark'
 describe Snapshot do
   describe Snapshot::TestCommandGenerator do
+    before do
+      fake_out_xcode_project_loading
+    end
```
Comment (mfurtak) [inline snapshot/spec/test_command_generator_spec.rb]: Is this necessary to do before every example or does `before(:all)` work?

File: snapshot/spec/spec_helper.rb [hunk 3]
```diff
@@ -10,3 +10,25 @@ module SpecHelper
...
+        Debug
+        Release
+
+    If no build configuration is specified and -scheme is not passed then "Release" is used.
+
+    Schemes:
+        Example
+        ExampleUITests
+EOS
+  allow_any_instance_of(FastlaneCore::Project).to receive(:raw_info).and_return fake_result
```
Comment (mfurtak) [inline snapshot/spec/spec_helper.rb]: I think we agree that this is a bit of a smell, and `rubocop-rspec` [seems to agree](https://github.com/nevir/rubocop-rspec/blob/master/lib/rubocop/cop/rspec/any_instance.rb#L6)... That said, this is a reasonable way of making this performant, and it probably has fewer gotchas than my instance caching way in `gym` and `scan`.

I think I would still prefer a `Project` data mocking approach that eliminates the `xctool` output and parsing entirely, but I'm on board with going this way for now, if you like.

**Comment by mfurtak** [comment]
This looks good to me with one nitpick!

**Comment by i2amsam** [comment]
Closing in favor of #3799

**Pull request closed without merging**